### DISCUSSION TITLE
Checking the platform running the Flet app

### DISCUSSION THREAD

**Discussion by ndonkoHenri** [flet-dev/flet#382] [Ideas]
Is there a way to check which platform is currently running the application we make? 
Ex: Browser, Win, Mac

Have a look at Flutter implementation for this...
[https://api.flutter.dev/flutter/dart-io/Platform-class.html](https://api.flutter.dev/flutter/dart-io/Platform-class.html)

**Comment by FeodorFitsner** [comment]
Sure, there three props for that:

* [`page.web`](https://flet.dev/docs/controls/page#web)
* [`page.platform`](https://flet.dev/docs/controls/page#platform)
* [`page.pwa`](https://flet.dev/docs/controls/page#pwa)

**Reply by ndonkoHenri** [reply]
How can I make my app run as a PWA? 
Any guide please?

**Reply by FeodorFitsner** [reply]
It's PWA already. You can customise app's `manifest.json` though: https://github.com/flet-dev/website/blob/chat-tutorial/docs/guides/python/deploying-web-app/progressive-web-apps.md

**Reply by ndonkoHenri** [reply]
![image](https://user-images.githubusercontent.com/98978078/192172913-ac36b85b-2c89-4eba-8523-dc1a5da2fc25.png)

I tried to follow the instructions in the file.
But can't see the "Install app..."

**Reply by FeodorFitsner** [reply]
What version of iOS is that?

**Reply by ndonkoHenri** [reply]
v15.5

**Reply by FeodorFitsner** [reply]
Alright, what browser is that? It must be Safari.

**Reply by ndonkoHenri** [reply]
No. 🙂
It's Chrome.

**Comment by FeodorFitsner** [comment]
iOS 16, Safari:

![image](https://user-images.githubusercontent.com/5041459/192190968-d4829e5c-18cf-4818-96c8-8b08a7ecfae0.png)

**Comment by FeodorFitsner** [comment]
https://superpwa.com/doc/test-pwa-ios-devices/#:~:text=Upon%20iOS%20devices%2C%20PWA%20currently%20works%20only%20upon%20Safari%20Browsers.%20And%20browsers%20like%20Chrome%20for%20iOS%2C%20Firefox%20Focus%20for%20iOS%2C%20Dolphin%20Ghostery%2C%20Opera%20Mini%20and%20Adblock%20Browser%20won%E2%80%99t%20support%20PWA%20at%20this%20moment.

**Reply by ndonkoHenri** [reply]
Very Interesting resource. ✅ 
It worked now, but only on safari.(as mentioned on the website)
🚀
![image](https://user-images.githubusercontent.com/98978078/192390518-194dcb3b-ab2b-4ede-880c-163c8832dd31.jpeg)

Is it possible to makes changes(customize) on the `manifest` file ? If yes ... How?

**Reply by ndonkoHenri** [reply]
😅 Just from seeing your previous response(about the `manifest.json` file): [https://github.com/flet-dev/flet/discussions/382#discussioncomment-3728311](https://github.com/flet-dev/flet/discussions/382#discussioncomment-3728311)

I will give it a deep check and let you know, if there are any difficulties.
